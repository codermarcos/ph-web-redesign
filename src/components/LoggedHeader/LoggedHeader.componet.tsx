import {
  HeaderContainer,
  IconMagnificGlass,
  Image,
  Input,
  SearchButton,
} from '@/components/LoggedHeader/LoggedHeader.styles';
import { useState } from 'react';

export type UserType = {
  name: string;
  profileImage: string;
}

function LoggedHeader({ name, profileImage }: UserType) {

  const [
    search,
    setSearch,
  ] = useState(false);

  return (
    <HeaderContainer>
      <Image
        alt={name}
        height={74}
        src={profileImage}
        width={74}
      />

      <Input type="search" />

      <SearchButton
        onClick={() => setSearch(!search)}
        type="button"
      >
        <IconMagnificGlass
          className={
            search
              ? 'active'
              : ''
          }
        />
      </SearchButton>
    </HeaderContainer>
  );

}

export default LoggedHeader;
