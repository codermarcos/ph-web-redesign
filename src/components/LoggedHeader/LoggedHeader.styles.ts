import styled from 'styled-components';

import NextImage from 'next/image';

import { padding, percent, px, rotateZ, translateX, translateY } from 'csx';

import { colors, sizes } from '@/styles/variables';

export const HeaderContainer = styled.section({
  display: 'flex',
  gridArea: 'header',
  justifyContent: 'space-between',
  maxWidth: sizes.container,
  padding: padding(sizes.space.default),
  width: percent(100),
});

export const Image = styled(NextImage)({
  borderRadius: percent(100),
});

export const Input = styled.input({
  backgroundColor: colors.whiteDarkness1,
  border: 'none',
  borderRadius: sizes.borderRadius.default,
});

export const IconMagnificGlass = styled.div({
  border: '4px solid black',
  borderRadius: percent(100),
  height: px(28),
  position: 'relative',
  transition: 'all ease-in .1s',
  width: px(28),

  // eslint-disable-next-line sort-keys
  '&.active': {
    height: px(38),
    width: px(38),

    // eslint-disable-next-line sort-keys
    ':after': {
      transform: `${translateX(px(-11))} ${translateY(px(13))} ${rotateZ('45deg')}`,
      width: px(22),
    },

    ':before': {
      opacity: 1,
      transform: `${translateX(px(-11))} ${translateY(px(13))} ${rotateZ('-45deg')}`,
    },
  },

  ':after': {
    backgroundColor: 'black',
    borderRadius: px(4),
    content: '""',
    height: px(4),
    position: 'absolute',
    transform: `${translateX(px(4))} ${translateY(px(24))} ${rotateZ('45deg')}`,
    transition: 'all ease-out .1s',
    width: px(20),
  },

  ':before': {
    backgroundColor: 'black',
    borderRadius: px(4),
    content: '""',
    height: px(4),
    opacity: 0,
    position: 'absolute',
    transform: `${translateX(px(-20))} ${translateY(px(16))} ${rotateZ('-45deg')}`,
    transition: 'all ease-in .1s',
    width: px(22),
  },
});

export const SearchButton = styled.button({
  alignItems: 'center',
  backgroundColor: 'transparent',
  border: 'none',
  borderRadius: percent(100),
  display: 'flex',
  justifyContent: 'center',
  width: px(74),
});

