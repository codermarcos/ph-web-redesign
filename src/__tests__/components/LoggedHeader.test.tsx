import { RenderResult, render } from '@testing-library/react';
import { faker } from '@faker-js/faker';

import LoggedHeader from '@/components/LoggedHeader/LoggedHeader.component';

faker.seed(1);

const user = {
  name: faker.name.fullName(),
  profileImage: faker.image.avatar(),
};

describe(
  '[Component] -> components/LoggedHeader',
  () => {

    // eslint-disable-next-line init-declarations
    let wrapper: RenderResult;

    beforeEach(
      () => {

        wrapper = render(
          <LoggedHeader {...user} />
        );

      }
    );

    it(
      'should render correct header',
      () => {

        expect(wrapper.container).toMatchSnapshot();

      }
    );

  }
);
