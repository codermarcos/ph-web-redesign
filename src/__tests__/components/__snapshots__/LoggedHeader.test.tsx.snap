// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] -> components/LoggedHeader should render correct header 1`] = `
<div>
  <section
    class="sc-a31e1884-0 bBqNYt"
  >
    <img
      alt="Winifred Reichel Sr."
      class="sc-a31e1884-1 eRMiTh"
      data-nimg="1"
      decoding="async"
      height="74"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fcloudflare-ipfs.com%2Fipfs%2FQmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye%2Favatar%2F377.jpg&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fcloudflare-ipfs.com%2Fipfs%2FQmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye%2Favatar%2F377.jpg&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fcloudflare-ipfs.com%2Fipfs%2FQmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye%2Favatar%2F377.jpg&w=256&q=75 2x"
      style="color: transparent;"
      width="74"
    />
    <input
      class="sc-a31e1884-2 bYKFjv"
      type="search"
    />
    <button
      class="sc-a31e1884-4 ktTJOl"
      type="button"
    >
      <div
        class="sc-a31e1884-3 fuKxSY"
      />
    </button>
  </section>
</div>
`;
